bin/upgrade_terraform_modules.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+function github_latest_release() {
+  local org=$1
+  local repo=$2
+  local ref=$(curl -sSL https://api.github.com/repos/$org/$repo/releases/latest | jq .tag_name -r)
```

Comment:
GitHub rate-limits unauthenticated calls to their api endpoint by source IP.

https://developer.github.com/changes/2012-10-14-rate-limit-changes/

I've found this kind of query works fine locally, but fails sporadically from CI. The typical workaround seems to be to create a bot user with zero access to anything, generate an access token, and then use basic auth in the curl command....

```
curl -sSL https://api.github.com/repos/$org/$repo/releases/latest?access_token=<the access token>
```

Reply:
You're right. I was just being lazy =)

I've added support for the tokens. Decided to use a header since that is less likely to show up in access logs. https://gist.github.com/caspyin/2288960#oauth